### PR TITLE
feat(Analyzer): Add option to Pub to skip native Flutter dependencies

### PR DESCRIPTION
**Pull request by maklostermann**
Flutter packages typically also include native dependencies for Android (Gradle) and iOS (CocoaPods). A new configuration option allows to limit the analyzer to the Pub dependencies only.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 1]
```diff
@@ -246,28 +260,32 @@ class Pub(
...
-                    val gradleDependencies = gradleDefinitionFiles.map {
-                        PackageManagerDependencyHandler.createPackageManagerDependency(
-                            packageManager = gradleFactory.type,
-                            definitionFile = VersionControlSystem.getPathInfo(it).path,
-                            scope = "releaseCompileClasspath",
-                            linkage = PackageLinkage.PROJECT_STATIC
+            if (!pubDependenciesOnly) {
+                val gradleDefinitionFiles = gradleDefinitionFilesForPubDefinitionFiles.getValue(definitionFile).toList()
+
+                if (!pubDependenciesOnly) {
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pub.kt:266]: Is checking for `!pubDependenciesOnly` twice intended?

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 2]
```diff
@@ -128,12 +131,19 @@ class Pub(
...
     override fun transformVersion(output: String) = output.removePrefix("Dart SDK version: ").substringBefore(' ')
 
     override fun getVersionRequirement(): RangesList = RangesListFactory.create("[2.10,)")
 
     override fun beforeResolution(definitionFiles: List<File>) {
-        if (gradleFactory != null) {
+        if (pubDependenciesOnly) {
+            logger.info {
+                "Only analyzing Pub dependencies, skipping additional package managers for Flutter packages" +
+                    " (Gradle for Android, CocoaPods for iOS dependencies)."
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pub.kt:144]: Nit: Please move the continuing space between these lines from the second to the first line, i.e. end the first line with `packages " +` and start the second line with `"(Gradle`.

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 3]
```diff
@@ -72,6 +72,8 @@ import org.ossreviewtoolkit.utils.ort.showStackTrace
 import org.semver4j.RangesList
 import org.semver4j.RangesListFactory
 
+private const val OPTION_PUB_DEPENDENCIES_ONLY = "pubDependenciesOnly"
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pub.kt:75]: Ah, this reminds my I should have added a constant for `"gradleVersion"`, too. Thanks 😁

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/Pub.kt (hunk 1 above)
Comment (maklostermann) [reply analyzer/src/main/kotlin/managers/Pub.kt]: Not really, something went wrong when rebasing.

File: analyzer/src/main/kotlin/managers/Pub.kt (hunk 2 above)
Comment (maklostermann) [reply analyzer/src/main/kotlin/managers/Pub.kt]: Okay, changed as suggested.

File: analyzer/src/main/kotlin/managers/Pub.kt (hunk 3 above)
Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/Pub.kt:75]: See https://github.com/oss-review-toolkit/ort/pull/6641.

File: analyzer/src/main/kotlin/managers/Pub.kt [hunk 4]
```diff
@@ -246,28 +260,31 @@ class Pub(
...
-                            scope = "releaseCompileClasspath",
-                            linkage = PackageLinkage.PROJECT_STATIC
-                        )
-                    }.toSortedSet()
+            if (!pubDependenciesOnly) {
+                if (gradleFactory != null) {
+                    val gradleDefinitionFiles = gradleDefinitionFilesForPubDefinitionFiles
+                        .getValue(definitionFile)
+                        .toList()
+                    if (gradleDefinitionFiles.isNotEmpty()) {
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Pub.kt:268]: Nit: There should now be an empty line before this one to separate from the wrapped block above.

**Review by sschuberth — CHANGES_REQUESTED** [review]
I'll leave the approval to our "Pub expert" @MarcelBochtler 😀

**Review by mnonnenmacher — APPROVED** [review]

**Pull request merged**